### DISCUSSION TITLE
CLI - New optional parameter to download the quicklook of products

### DISCUSSION THREAD

**Discussion by ahuarte47** [CS-SI/eodag#263] [Ideas]
Congratulations for your amazing software!

It would great that `search` command parameter of CLI tool had a new option to download the quicklook of a product. It could work in colaboration of current `--storage` parameter. Maybe something similar to `--quicklook` in [sentinelsat](https://github.com/sentinelsat/sentinelsat/blob/821e6d03acb0bc9390a56d59f73ea06d62e09689/sentinelsat/scripts/cli.py#L132) module.

Product class already has this [method](https://github.com/CS-SI/eodag/blob/3c7598b22ba9e3de1c6ed901455b42b65179e5d9/eodag/api/product/_product.py#L284),

Can be this new option interesting?

**Comment by sbrunato** [comment]
Hi, and many thanks!

This new option for the CLI is indeed interesting. What would you think of such an implementation, that would download quicklooks instead of products:
```bash
eodag download --quicklooks --search-results search_results.geojson
```

**Reply by ahuarte47** [reply]
Thanks @sbrunato ! You are right, it seems more coherent adding this new parameter for `download` command instead for `search` command. Although when a user is analizying results of available products (For instance, what areas of his results are cloudy), he has to execute two commands. It is not a "problem" for me, but I am thinking about the more common way of an user would use this. Of course, it is up to you. Thank you very much!

**Reply by sbrunato** [reply]
The method we already implement in the Python API [EOProduct.get_quicklook](https://eodag.readthedocs.io/en/latest/api_reference/eoproduct.html#eodag.api.product._product.EOProduct.get_quicklook) works on a product, or a search result.

So, to have a similar functionality in the CLI and be coherent, we should execute the command on an existing search result like:

> ```shell
> eodag download --quicklooks --search-results search_results.geojson
> ```

or maybe better:
```
eodag get_quicklook --search-results search_results.geojson
```

**Reply by maximlt** [reply]
That's a very good idea!

I'm not so much in favor of adding this option to `eodag search`, since we already have a `download` sub-command that's in charge of downloading stuff. I think I prefer `eodag download --quicklooks --search-results search_results.geojson`.
